lib/Frenetic_Fdd.ml

```diff
@@ -65,34 +66,24 @@ module Field = struct
     | Location -> "Location"
     | VFabric -> "VFabric"
 
-  let num_fields = 15
-
-  (* Ensure that these are in the same order in which the variants appear. *)
-  let all_fields =
-    [ Switch; Vlan; VlanPcp; VSwitch; VPort; EthType; IPProto; EthSrc; EthDst;
-      IP4Src; IP4Dst; TCPSrcPort; TCPDstPort; Location; VFabric]
-
   let is_valid_order (lst : t list) : bool =
-    List.length lst = num_fields &&
-    List.for_all all_fields ~f:(List.mem lst)
-
-  (* assert (is_valid_order all_fields) *)
+    Set.Poly.(equal (of_list lst) (of_list all))
 
   (* Initial order is the order in which fields appear in this file. *)
   let order = Array.init num_fields ~f:ident
 
```

Comment:
I was able to remove this imperative code. No more danger of having inconsistencies between `readable_order` and `order`.